src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -440,5 +467,36 @@ private IGraphType ConvertTypeReference(INamedType parentType, IGraphType type)
 
             return result;
         }
+
+        private void SetGraphType(string typeName, IGraphType type)
+        {
+            if (string.IsNullOrWhiteSpace(typeName))
+            {
+                throw new ArgumentOutOfRangeException(nameof(typeName), "A type name is required to lookup.");
+            }
+
+            if (_types.TryGetValue(typeName, out var existingGraphType))
+            {
+                if (ReferenceEquals(existingGraphType, type))
+                {
+                    // nothing to do
+                }
+                else if (existingGraphType.GetType() == type.GetType())
+                {
+                    _types[typeName] = type; // this case worked before overwriting the old value
+                }
+                else
+                {
+                    throw new InvalidOperationException(
+                        $@"You are trying to register a second GraphType '{type.GetType().FullName}' with the name '{typeName}'.
```

Comment:
Quite a long message. Correct me if something is wrong. I am not a native speaker.

Reply:
I like the verbosity of the message, but the part of speech may be a bit awkward. Perhaps something like:

> Unable to register GraphType '{type.GetType().FullName}' with the name '{typeName}'; the name '{typeName}' is already registered to '{existingGraphType.GetType().FullName}'.